x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ssl/CertParsingUtils.java

```diff
@@ -93,6 +95,23 @@ public static KeyStore readKeyStore(Path path, String type, char[] password)
         return certificates.toArray(new Certificate[0]);
     }
 
+    public static X509Certificate readX509Certificate(Path path) throws CertificateException, IOException {
+        List<Certificate> certificates = PemUtils.readCertificates(List.of(path));
+        if (certificates.size() != 1) {
+            throw new IllegalArgumentException("expected a single certificate in file [" + path.toAbsolutePath() + "] but found [" +
+                certificates.size() + "]");
+        }
+        final Certificate cert = certificates.get(0);
+        if (cert instanceof X509Certificate) {
+            return (X509Certificate) cert;
+        } else {
+            throw new IllegalArgumentException("the certificate in " + path.toAbsolutePath() + " is not an X.509 certificate ("
+                + cert.getType()
+                + " : "
+                + cert.getClass() + ")");
+        }
+    }
+
```

Comment:
This isn't exactly in scope for this PR, but a test case was using the new method so it made sense to add it now..